tests/functional/test_adcm_upgrade.py

```diff
@@ -0,0 +1,173 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=W0621,R0914
+
+import allure
+import pytest
+
+from contextlib import contextmanager
+from docker.errors import NotFound
+
+from adcm_client.objects import ADCMClient
+from adcm_pytest_plugin.utils import (get_data_dir, random_string, get_initialized_adcm_image)
+from adcm_pytest_plugin.plugin import parametrized_by_adcm_version
+from adcm_pytest_plugin.docker import DockerWrapper
+
+
+def old_adcm_images():
+    # TODO Change oldest version after https://arenadata.atlassian.net/browse/ADCM-1430
+    # return parametrized_by_adcm_version(adcm_min_version="2019.01.30")[0]
+    return parametrized_by_adcm_version(adcm_min_version="2019.03.18")[0]
+
+
+@contextmanager
+def adcm_client(adcm_repo, adcm_tag, volumes, pull=False):
+    """
+    Run ADCM container from image {adcm_repo}:{adcm_tag}. If image was pulled remove it after
+    """
+    if pull:
+        with allure.step(f"Create isolated image copy from {adcm_repo}:{adcm_tag}"):
+            new_image = get_initialized_adcm_image(adcm_repo=adcm_repo,
+                                                   adcm_tag=adcm_tag,
+                                                   pull=True)
+            repo, tag = new_image["repo"], new_image["tag"]
+    else:
+        repo, tag = adcm_repo, adcm_tag
+    with allure.step(f"Run ADCM from image {repo}:{tag}"):
+        dw = DockerWrapper()
+        adcm = dw.run_adcm(image=repo, tag=tag, volumes=volumes, pull=False)
+        adcm.api.auth(username='admin', password='admin')
```

Comment:
Good idea! I will create task to move all hardcoded credentials to separate fixtures. Won't fix here since it's out of PR scope. https://arenadata.atlassian.net/browse/ADCM-1435